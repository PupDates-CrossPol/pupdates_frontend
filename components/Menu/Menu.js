import * as React from 'react';
import { StyleSheet, Text, View, Image, TextInput, Button, TouchableOpacity, ScrollView, SafeAreaView } from 'react-native';
import { createAppContainer, createSwitchNavigator } from 'react-navigation';
import { createStackNavigator } from 'react-navigation-stack';
import HomeScreen from '../Home/Home'
import { Ionicons } from '@expo/vector-icons';
import { connect } from 'react-redux'

export class MenuScreen extends React.Component {
    static navigationOptions = ({navigation}) => ({        
        headerLeft: () => <Image source={require('../../assets/PupDatesLogo.png')} style={styles.logo} />,
        headerTitle: () => <Image source={require('../../assets/MenuSpread.png')} style={styles.navTitle}/>,
        headerRight: () => (
            <TouchableOpacity onPress={() => navigation.navigate('Home')}>
                <Ionicons name="ios-close" size={50} color='rgb(21, 112, 125)' />
            </TouchableOpacity>
        ),
        headerStyle: {
          backgroundColor: '#fff',
        },
        headerTintColor: '#fff',
        headerTitleStyle: {
          fontWeight: 'bold',
        },
        headerLeftContainerStyle: styles.leftNavIcon,
        headerRightContainerStyle: styles.rightNavIcon,
      })

    render() {
        return (
                <View style={styles.container}>
                    <View style={styles.menuOptions}>
                      <TouchableOpacity onPress={() => this.props.navigation.navigate('Matches')} style={styles.menuImgText}>
                        <Image source={require('../../assets/PupDatesLogo.png')} style={styles.menuCircle} />
                        <Text style={styles.menuOptionsText} >Matches</Text>
                      </TouchableOpacity>
                    </View>
                    <View style={styles.menuOptions}>
                      <TouchableOpacity onPress={() => this.props.navigation.navigate('UserProfile')} style={styles.menuImgText}>
                        <Image source={{uri: this.props.user.image}} style={styles.menuCircle} />
                        <Text style={styles.menuOptionsText} >{this.props.user.first_name}</Text>
                      </TouchableOpacity>
                    </View>
                    <View style={styles.menuOptionsBottom}>
                      <TouchableOpacity onPress={() => this.props.navigation.navigate('DogPack')} style={styles.menuImgText}>
                        <Image source={{uri: this.props.packPhotos[0].image_url}} style={styles.menuCircle} />
                        <Text style={styles.menuOptionsText} >Dog Pack</Text>
                      </TouchableOpacity>
                    </View>
                </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
      flex: 1,
      flexDirection: 'column',
      backgroundColor: '#fff',
      alignItems: 'center',
      justifyContent: 'space-between',
      marginTop: '2%',
      marginBottom: '2%'
    },
    menuOptions: {
      flex: 1,
      flexDirection: 'column',
      backgroundColor: '#fff',
      alignItems: 'center',
      justifyContent: 'center',
      borderBottomColor: 'rgba(0,0,0,0.57)',
      borderBottomWidth: 1,
      width: '75%',
    },
    menuOptionsBottom: {
      flex: 1,
      flexDirection: 'column',
      backgroundColor: '#fff',
      alignItems: 'center',
      justifyContent: 'center',
      width: '75%',
    },
    menuImgText: {
      flexDirection: 'column',
      backgroundColor: '#fff',
      alignItems: 'center',
      paddingTop: 10,
      width: '100%',
    },
    menuOptionsText: {
      fontSize: 25,
      fontWeight: '300',
      marginTop: 5,
    }, 
    menuOptionsTextSub: {
      color: 'rgba(0,0,0,0.57)',
      fontSize: 25,
      fontWeight: '300',
    },
    menuCircle: {
      aspectRatio: 1/1,
      height: '65%',
      borderRadius: 100,
      borderColor: 'black',
      borderWidth: 2,
    },
    logo: {
        height: 40,
        width: 40,
    }, 
    image: {
      height: 140,
      width: 140,
    },
    title: {
      fontSize: 50,
      // fontFamily: 'major-mono-display'
    },
    input: {
      height: 30,
      width: '70%',
      borderColor: 'lightgrey',
      borderRadius: 50,
      // borderColor: 'rgba(33,33,33,0.81)',
      borderWidth: 1.5,
      padding: 10,
      height: 60,
      margin: 20,
      alignItems: 'center'
    },
    linearGradient: {
      width: '70%',
      borderRadius: 25,
      height: 50,
      alignItems: 'center',
      justifyContent: 'center',
    },
    button: {
      // borderRadius: 50,
      borderTopRightRadius: 20,
      borderBottomRightRadius: 20,
      borderBottomLeftRadius: 20,
      borderTopLeftRadius: 20,
      alignItems: 'center',
      justifyContent: 'center',
      color: 'black'
    },
    buttonText: {
      color: '#fff',
      fontSize: 25,
    },
    navTitle: {
        width: 160,
        height: 40,
        marginBottom: 5,
    },
    rightNavIcon: { 
        marginRight: 20,  
        marginBottom: 10, 
    },
    leftNavIcon: { marginLeft: 10,  marginBottom: 5, }
  });


  export const mapStateToProps = state => ({
    user: state.user,
    packPhotos: state.packPhotos
  })

  export default connect (mapStateToProps)(MenuScreen)
// const AppNavigator = createStackNavigator({
//     Menu: {
//         screen: MenuScreen,
//     },
// });
//   export default createAppContainer(AppNavigator)

// export default MenuScreen;