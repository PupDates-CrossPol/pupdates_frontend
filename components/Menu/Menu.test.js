import React from 'react'
import renderer from 'react-test-renderer'
import { MenuScreen as Menu, mapStateToProps, mapDispatchToProps } from './Menu'

describe('<Menu />', () => {
  it('has 4 children', () => {
    const mockUser = {
      id: 1,
      first_name: 'bob',
      last_name: 'ross',
      description: 'happy little trees',
      email: 'happylittletrees@email.com',
      image: 'an image url'
    }

    const mockPackPhotos = [
      {
        id: 1,
        dog_id: 1,
        image: 'somestring'
      }
    ]

    const naviProp = { navigation: { navigate: () => {} } };

    const tree = renderer.create(<Menu user={mockUser} packPhotos={mockPackPhotos} navigation={naviProp} />).toJSON()

    expect(tree.children.length).toBe(3)
  })

  it('renders correctly', () => {
    const mockUser = {
      id: 1,
      first_name: 'bob',
      last_name: 'ross',
      description: 'happy little trees',
      email: 'happylittletrees@email.com',
      image: 'an image url'
    }

    const mockPackPhotos = [
      {
        id: 1,
        dog_id: 1,
        image: 'somestring'
      }
    ]

    const naviProp = { navigation: { navigate: () => {} } };

    const tree = renderer.create(<Menu user={mockUser} packPhotos={mockPackPhotos} navigation={naviProp} />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('mapStateToProps', () => {
  it('should return an object with the user object', () => {
    const mockState = {
      user: {
        id: 1,
        first_name: 'sara',
        last_name: 'karsh',
        email: 'email',
        description: 'a string of words'
      },
      swipePack: [
        {
          id: 1,
          user_id: 1,
          name: 'fluffy',
        }
      ],
      swipePackPhotos: [
        {
          id: 1,
          dog_id: 1,
          image_url: 'someaddress'
        }
      ]
    }

    const expectedState = {
      user: {
        id: 1,
        first_name: 'sara',
        last_name: 'karsh',
        email: 'email',
        description: 'a string of words'
      },
    }

    const mappedProps = mapStateToProps(mockState)

    expect(mappedProps).toEqual(expectedState)
  })
})