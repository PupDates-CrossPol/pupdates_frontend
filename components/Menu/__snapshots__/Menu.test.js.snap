// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-between",
      "marginBottom": "2%",
      "marginTop": "2%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderBottomColor": "rgba(0,0,0,0.57)",
        "borderBottomWidth": 1,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "width": "75%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flexDirection": "column",
          "opacity": 1,
          "paddingTop": 10,
          "width": "100%",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "aspectRatio": 1,
            "borderColor": "black",
            "borderRadius": 100,
            "borderWidth": 2,
            "height": "65%",
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 25,
            "fontWeight": "300",
            "marginTop": 5,
          }
        }
      >
        Matches
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderBottomColor": "rgba(0,0,0,0.57)",
        "borderBottomWidth": 1,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "width": "75%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flexDirection": "column",
          "opacity": 1,
          "paddingTop": 10,
          "width": "100%",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "an image url",
          }
        }
        style={
          Object {
            "aspectRatio": 1,
            "borderColor": "black",
            "borderRadius": 100,
            "borderWidth": 2,
            "height": "65%",
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 25,
            "fontWeight": "300",
            "marginTop": 5,
          }
        }
      >
        bob
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "width": "75%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flexDirection": "column",
          "opacity": 1,
          "paddingTop": 10,
          "width": "100%",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "aspectRatio": 1,
            "borderColor": "black",
            "borderRadius": 100,
            "borderWidth": 2,
            "height": "65%",
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 25,
            "fontWeight": "300",
            "marginTop": 5,
          }
        }
      >
        Dog Pack
      </Text>
    </View>
  </View>
</View>
`;
